### ISSUE TITLE
Resizing an animated WebP throws and error vips2webp: unable to encode

### ISSUE THREAD

**Issue by metju** [lovell/sharp#2501]
I'm trying to resize some files, but it starts failing very soon. Might be hitting some WebP file dimension limit? I've noticed when dealing with animated gifs, it loads all the pages into one tall file when processing.

```const sharp = require('sharp')

sharp('animated-webp-supported.webp', {animated: true})
	.resize(2880)
	.webp()
	.toFile('resized.webp')
```

[the file I'm trying to convert](https://github.com/lovell/sharp/files/5750456/test.zip) (zipped, since GitHub doesn't like WebPs)

**Comment by lovell** [comment]
Hi, given you're resizing, you'll need to update the `pageHeight`, in this 12-page example to 2880 / 12 = 240.

```diff
-  .webp()
+  .webp({ pageHeight: 240 })
```

https://sharp.pixelplumbing.com/api-output#webp

See also https://github.com/lovell/sharp/issues/2275 for a future possible improvement relating to this.

**Comment by metju** [comment]
Hello, thanks for the response!

240 didn't work, but 2880 did (it's the new height of the resized image, right?). I've tried setting the `pageHeight` before asking, but I didn't guess the right value.

Is it possible to get the new exact height of a resized image? If not, and I have to calculate it myself, should I `floor`, `ceil` or `round` the value? I've tried the conversion with `2879` and `2881` and both didn't work, so this seems important to get right.

When setting the `pageHeight`, are there more than one correct value, that wouldn't produce the `unable to encode` error? Could then `sharp` do this work for us and calculate it?
